Add total tally

vladh opened the pull request:
Closes #186.

Opening this PR as a place to discuss ideas for implementation details.

Potential issues with this:

* Wordsmithing would be nice.
* We probably want a bit more design work here.
* I think the current phrasing is clear, but maybe it would be worth adding a footnote saying that this is the total of all annual reports of all members we're tracking, including donations made before the Pledge existed.

![2024-10-08-16-23-22](https://github.com/user-attachments/assets/70439f8a-e017-4858-9bed-1d5550b511f1)

vladh commented:
Iterating on this.

![2024-10-10-14-19-24](https://github.com/user-attachments/assets/130f4747-d3b1-4aa8-8ff4-1bc0046115b8)
![2024-10-10-14-19-40](https://github.com/user-attachments/assets/5436855c-5d16-4c68-808b-9bb311737017)

I think we're getting somewhere, but we're going to need a bit of a visual tidy up first, things are getting too busy/messy.

vladh commented:
![2024-10-10-14-26-17](https://github.com/user-attachments/assets/77566618-9300-4e06-806e-6f7af421ff5e)

vladh commented:
@chadwhitacre This is ready for review. I think it's good enough to iterate on later.

![2024-10-11-17-40-15](https://github.com/user-attachments/assets/470aa975-51e8-4f13-b4d7-a36f48ebed32)

![2024-10-11-17-40-21](https://github.com/user-attachments/assets/ff7857a5-6444-4075-b847-d61ba594100a)

chadwhitacre commented on src/pages/about.astro:
```diff
@@ -216,6 +218,15 @@ const members = filterInactiveMembers(await getMembers());
+        <p
+          >The grand total is calculated by adding up all of the payments made by our members across all of their annual
+          reports. Note that some of our members' payments may precede their joining the Pledge.</p
+        >
```
Can we do latest annual report?

chadwhitacre replied:
Latest annual report that is newer than a year old. ideally.

vladh commented:
@chadwhitacre I've implemented this. I do wonder about something, though. Because this grand total is rolling, it prevents us from displaying how much money we've raised in total. The total is always over some not-clearly-defined period.

Showing a _total_ total amount might be very useful for marketing. For example, we were saying it would be nice to pursue certain milestones, where media coverage might be relevant when we reach eg $5 million.

chadwhitacre commented:
Hmm ... fair points ... 🤔

I guess my thinking is that, in the same way that we expect companies to pay every year, we should report on a yearly basis. What have you done for me lately? A graph that only ever goes up and to the right is generally low-value information. What if report yearly amounts along these lines?

Q1 - "Last year our members paid maintainers"
Q2-Q4 - "So far this year our members paid maintainers"

chadwhitacre commented:
We could show the previous period in small text beneath for comparison (fine if out of scope here).

vladh commented:
Hmm…the vast majority of our members have the end of the year as their dateYearEnding. So “total given this year” might be strongly skewed towards $0, as most members will report towards the end of the year.

chadwhitacre commented:
As long as the report is filed with us it's fine if the year end is in the future. No?

vladh commented:
Of course, I'm just thinking that members might not report until later in the year, and the number on the website might be quite low for a good part of the year.

Going to think about this more…

chadwhitacre commented:
We could also manually compute a number and update it periodically. Make it static instead of dynamic, in other words.

vladh commented:
That's true, but we would still have to decide how to calculate the number.

Still not sure what the best way forward here is. It seems that some grand all-time total is useful, because it will show us, and help us celebrate, the milestones we've reached — 2 million, 5 million, 10 million…

On the other hand, whether we want to do yearly totals is a valid question.

I've written this tool to visualise our members' reporting periods: [reportviz][reportviz].

Here is a screenshot:

![2024-10-12-12-40-35](https://github.com/user-attachments/assets/16776edb-2435-4527-8570-fffd054475f7)

18 out of 25 members report at the end of the year.

[reportviz]: https://github.com/opensourcepledge/osp-tools/tree/main/reportviz

vladh commented:
Reverted ["TotalTally: only take payments from the latest annual report"](https://github.com/opensourcepledge/opensourcepledge.com/pull/188/commits/5e45211013a96ed7ef7c1daa1b6a7b9b30092ae0), because the numbers it yields are not really possible to interpret.

chadwhitacre commented:
> It seems that some grand all-time total is useful, because it will show us, and help us celebrate, the milestones we've reached — 2 million, 5 million, 10 million…

Sentry [celebrates ARR milestones](https://blog.sentry.io/building-for-the-fortune-500-000/) (our previous [engagement with Nasdaq](https://www.linkedin.com/posts/getsentry_big-day-thanks-to-our-friends-at-nasdaq-activity-7196592371327787008-O7FP/) ;-). Presumably other companies do as well. I think we'll be more credible if we do the same for Pledge, because it's a more meaningful measurement of impact of the Pledge on the overall vitality of the Open Source ecosystem than total sum of revenue.

Sentry can effectively be classed with Tideways and Private Packagist as having reported on last year, because I reported based on our fiscal year, which ends on January 31. Let's reach out to [Tideways](https://github.com/opensourcepledge/crm/issues/13) and [Private Packagist](https://github.com/opensourcepledge/crm/issues/14). If they are planning on making 2024 payments before EOY as Sentry is, then we effectively have:

- 2023 - 3 reports
- 2024 - 22 reports, with 3 more expected from current members before EOY

I've expanded the CRM to track [renewals](https://github.com/opensourcepledge/crm/issues?q=is%3Aissue+is%3Aopen+label%3Arenewal).

We may need to provide guidance on report timing on the FAQ, and possibly iterate on the JSON format. It seems year ending is maybe less important than renewal date.

vladh commented:
I rebased this, and updated it to only include members' latest report:

![2024-11-13-19-59-47](https://github.com/user-attachments/assets/67f76bc6-ec53-4946-b58f-99f4dddd755b)

This is just to keep things up to date — we might still want to discuss/change this further.
